lib/octokit/client/deployments.rb

```diff
@@ -1,69 +1,87 @@
-      # @return [Sawyer::Resource] A deployment
+      # @param ref [String] The ref to deploy. This can be a branch, tag, or SHA.
+      # @param options [String] :task Specifies a task to execute (e.g., `deploy` or `deploy:migrations`).
+      # @param options [Boolean] :auto_merge Attempts to automatically merge the default branch into the requested ref, if it is behind the default branch.
+      # @param options [Array<string>] :required_contexts The status contexts to verify against commit status checks. If this parameter is omitted, then all unique contexts will be verified before a deployment is created. To bypass checking entirely pass an empty array. Defaults to all unique contexts.
+      # @param options [String] :payload JSON payload with extra information about the deployment.
+      # @param options [String] :environment Name for the target deployment environment (e.g., `production`, `staging`, `qa`).
+      # @param options [String] :description Short description of the deployment.
+      # @param options [Boolean] :transient_environment Specifies if the given environment is specific to the deployment and will no longer exist at some point in the future. **This parameter requires a custom media type to be specified.**
+      # @param options [Boolean] :production_environment Specifies if the given environment is one that end-users directly interact with. **This parameter requires a custom media type to be specified.**
+      # @return <Sawyer::Resource> The new deployment
       # @see https://developer.github.com/v3/repos/deployments/#create-a-deployment
       def create_deployment(repo, ref, options = {})
         options[:ref] = ref
         post("#{Repository.path repo}/deployments", options)
       end
 
-      # List all statuses for a Deployment
+      # Get a single deployment status
+      #
+      # @param deployment_url [String] A URL for a deployment resource.
+      # @param status_id [Integer] The deployment status ID.
+      # @return <Sawyer::Resource> A single deployment status
+      # @see https://developer.github.com/v3/repos/deployments/#get-a-single-deployment-status
+      def deployment_status(deployment_url, status_id, options = {})
+        deployment = get(deployment_url, accept: options[:accept])
+        get(deployment.rels[:statuses].href, options)
+      end
+
+      # List deployment statuses
       #
-      # @param deployment_url [String] A URL for a deployment resource
+      # @param deployment_url [String] A URL for a deployment resource.
       # @return [Array<Sawyer::Resource>] A list of deployment statuses
       # @see https://developer.github.com/v3/repos/deployments/#list-deployment-statuses
       def deployment_statuses(deployment_url, options = {})
-        deployment = get(deployment_url, :accept => options[:accept])
+        deployment = get(deployment_url, accept: options[:accept])
         get(deployment.rels[:statuses].href, options)
```

Comment:
This one was a bit tricky since it uses the rel rather than putting together the URL from the parameters defined in the routes file. In other words, I have done gross, gross things in order to support this. The TL;DR of the grossness is: (a) I override the params to add the `deployment_url` and delete the ID param and the repo param, and (b) I have a conditional that switches on whether it's a subresource or not, to determine what sort of method implementation to generate. I am almost 100% certain that this is going to turn out to not be good enough, because I would bet that we are not consistent in our choice of implementation. But for now this works.